src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +146,159 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
     def delete(self) -> None:
         """Deletes this Curated Hub"""
         return self._sagemaker_session.delete_hub(self.hub_name)
+
+    def _is_invalid_model_list_input(self, model_list: List[Dict[str, str]]) -> bool:
+        """Determines if input args to ``sync`` is correct."""
+        for obj in model_list:
+            if not isinstance(obj.get("model_id"), str):
+                return True
+            if "version" in obj and not isinstance(obj["version"], str):
+                return True
+        return False
+
+    def sync(self, model_list: List[Dict[str, str]]):
```

Comment:
this func will need to be split up, otherwise it's going to be near impossible to unit test properly